docs/terms/metric.md

```diff
@@ -4,12 +4,104 @@ title: Metric
+
+#### Types of MetricProvider Functions
+
+The diagram below shows the relationship between different types of MetricProvider functions.
+
+![Image](../images/metricprovider.png)
+
+You can view the implementations of MetricProvider subclasses in the `great_expectations/expectations/metrics` folder of a Great Expectations deployment, or [in the same folder of our GitHub](https://github.com/great-expectations/great_expectations/tree/develop/great_expectations/expectations/metrics).
+
+### Configuration
+
+Configuration of Metrics is applied when they are defined as part of an Expectation.
+
+#### Metrics naming conventions
+
+Metrics can have any name. However, for the "core" Great Expectations Metrics, we use the following conventions:
+
+* For **aggregate Metrics**, such as the mean value of a column, we use describe the domain and name of the statistic, such as `column.mean` or `column.max`.
+* For **map Metrics**, which produce values for individual records or rows, we define the domain using the prefix "column_values" and use several consistent suffixes to provide related Metrics. For example, for the Metric that defines whether specific column values fall into an expected set, several related Metrics are defined:
+    * `column_values.in_set.unexpected_count` provides the total number of unexpected values in the domain.
+    * `column_values.in_set.unexpected_values` provides a sample of unexpected_values; "result_format" is one of its
+      value_keys to determine how many values should be returned.
+    * `column_values.in_set.unexpected_rows` provides full rows for which the value in the domain column was unexpected
+    * `column_values.in_set.unexpected_value_counts` provides a count of how many times each unexpected value occurred
+
+Additionally, to facilitate optimized computation of Metrics, we use **Metric Partials** which define partially-parameterized functions that are necessary to build a desired Metric.
 
-NOTES: Everything under here is temporary, for use in writing the detailed page.
-----------
-Some attribute of data, such as the mean of a column. Metrics are generated and stored as part of running Expectations against a Batch (and can be referenced as such). For example, if you have an Expectation that the mean of a column fall within a certain range, the mean of the column must first be computed to see if its value is expected.
\ No newline at end of file
+* For aggregate Metrics, we often use an `ExecutionEngine` specific function with the suffix `.aggregate_fn`, such as `column.max.aggregate_fn`.
+* For map Metrics, to compute `column_values.in_set.unexpected_count`, we will rely on a **condition** called `column_values.in_set.condition`.
```

Comment:
This could be pretty confusing at this point.

I think there are two different things here which it's important  to differentiate:
- Metric implementations - this is where registration comes in, and where knowing about `aggregate_fn` etc would come in. This feels somewhat advanced, and would mostly come into play when implementing a new Expectation/Metric.
- Metric values - These are the metrics that have been calculated, have been stored in Validation Results, and could be accessed using Evaluation Parameters.

We may want a bit of discussion about Metric implementations here, but there's a lot of detail and complexity which could be overwhelming if a user is coming to this page to understand how to get `unexpected_values`, for instance. I could see having an "Advanced" header, and placing this below that on this page, but then I think a lot more explanation will be necessary.

Reply:
I've pulled this section and added it to the JIRA ticket with things to include in a document about creating new metrics.